src/research_index_backend/get_metadata.py

```diff
@@ -1,91 +1,81 @@
 from json import JSONDecodeError, dump
 from logging import DEBUG, basicConfig, getLogger
 from os import makedirs
+from typing import Dict
 
 import requests
 import requests_cache
 
 from .config import config
 
-logger = getLogger(__name__)
-basicConfig(
-    filename="research_index_backend.log",
-    filemode="w",
-    encoding="utf-8",
-    level=DEBUG,
-)
-
-
-def get_metadata_from_openaire(
-    session: requests_cache.CachedSession, doi: str, token
-):
-    """Gets metadata from OpenAire
+class MetadataFetcher:
+    def __init__(self, session: requests_cache.CachedSession, token: str = None):
+        self.session = session
+        self.token = token or config.token
+        self.logger = getLogger(__name__)
+        basicConfig(
+            filename="research_index_backend.log",
+            filemode="w",
+            encoding="utf-8",
+            level=DEBUG,
+        )
+
+    def _save_json_response(self, response, directory: str, doi: str) -> None:
```

Comment:
This is really only useful for debugging. In production, this will add overhead and use up disk space. I suggest adding a "debug mode" flag to disable this writing of JSON files by default.

Reply:
E.g. add `--write-metadata-to-json` flag into the CLI